tests/testTriggers.go

```diff
@@ -3126,6 +3131,167 @@ func TriggerUpgradeStork(contexts *[]*scheduler.Context, recordChan *chan *Event
+						"--auto-fstrim": "on",
+					})
+					if err != nil {
+						err = fmt.Errorf("error while enabling auto fstrim, Error:%v", err)
+						logrus.Error(err)
+						UpdateOutcome(event, err)
+
+					}
+					logrus.Info("AutoFsTrim is successfully enabled")
+					isAutoFsTrimEnabled = true
+					time.Sleep(5 * time.Minute)
+				} else {
+					logrus.Info("AutoFsTrim is already enabled")
+				}
+
+			})
+		Step("Validate AutoFsTrim Status ",
+			func() {
+				validateAutoFsTrim(contexts, event)
+
+			})
+
+		Step("Reboot attached node and validate AutoFsTrim Status ",
+			func() {
+
+				for _, ctx := range *contexts {
+					if strings.Contains(ctx.App.Key, "fstrim") {
+						vols, _ := Inst().S.GetVolumes(ctx)
+						for _, vol := range vols {
+
+							n, err := Inst().V.GetNodeForVolume(vol, 1*time.Minute, 5*time.Second)
+							UpdateOutcome(event, err)
+							logrus.Infof("volume %s is attached on node %s [%s]", vol.ID, n.SchedulerNodeName, n.Addresses[0])
+							err = Inst().S.DisableSchedulingOnNode(*n)
+							UpdateOutcome(event, err)
+
+							Inst().V.StopDriver([]node.Node{*n}, false, nil)
+
+							Inst().N.RebootNode(*n, node.RebootNodeOpts{
+								Force: true,
+								ConnectionOpts: node.ConnectionOpts{
+									Timeout:         1 * time.Minute,
+									TimeBeforeRetry: 5 * time.Second,
+								},
+							})
+
+							logrus.Info("wait for a minute for node reboot")
+							time.Sleep(1 * time.Minute)
+							n2, err := Inst().V.GetNodeForVolume(vol, 1*time.Minute, 5*time.Second)
+							if err != nil {
+
+								logrus.Info("Got error while getting node for volume %v, wait for 2 minutes to retry. Error: %v", vol.ID, err)
+								n2, _ = Inst().V.GetNodeForVolume(vol, 3*time.Minute, 10*time.Second)
+
+							}
+							logrus.Infof("volume %s is now attached on node %s [%s]", vol.ID, n2.SchedulerNodeName, n2.Addresses[0])
+							StartVolDriverAndWait([]node.Node{*n})
+							Inst().S.EnableSchedulingOnNode(*n)
+
+						}
+
+					}
+
+				}
+
+				validateAutoFsTrim(contexts, event)
+
+			})
+	})
+
+}
+
+func validateAutoFsTrim(contexts *[]*scheduler.Context, event *EventRecord) {
+	for _, ctx := range *contexts {
+		var appVolumes []*volume.Volume
+		var err error
+		if strings.Contains(ctx.App.Key, "fstrim") {
+			appVolumes, err = Inst().S.GetVolumes(ctx)
+			UpdateOutcome(event, err)
+			if len(appVolumes) == 0 {
+				UpdateOutcome(event, fmt.Errorf("found no volumes for app %s", ctx.App.Key))
+			}
+
+			for _, v := range appVolumes {
+				logrus.Infof("Getting info : %s", v.ID)
+				appVol, err := Inst().V.InspectVolume(v.ID)
+				if err != nil {
+					logrus.Errorf("Error inspecting volume: %v", err)
+				}
+				attachedNode := appVol.AttachedOn
+
+				fsTrimStatuses, err := Inst().V.GetAutoFsTrimStatus(attachedNode)
+				if err != nil {
+					UpdateOutcome(event, err)
+				}
+
+				val, ok := fsTrimStatuses[appVol.Id]
+				doExit := false
+				exitCount := 50
+
+				for !ok && !doExit {
```

Comment:
Put line: 3257 - 3286 into it's own function so that the code is more easily readable and we are separating the validation of fstrim per volumes logic.